src/components/annual-report/report-data/SectionOne.js

```diff
@@ -77,64 +87,300 @@ const reportQueries = {
+        dataSets,
+        dataElements,
+        orgUnits,
+        orgUnitLevel,
+        periods,
+        currentPeriod,
     }
 
+    useEffect(() => {
+        if (previewReport) {
+            refetch(variables)
+        }
+    }, [previewReport])
+
     if (loading) {
-        return <span>loading</span>
+        return (
+            <div
+                style={{
+                    display: 'flex',
+                    justifyContent: 'center',
+                    marginTop: '50px',
+                }}
+            >
+                {' '}
+                {/* TODO: put styles external */}
+                <CircularLoader large />
+            </div>
+        )
     }
 
     if (error) {
         return <span>error</span>
     }
 
     if (data) {
-        return <span>{JSON.stringify(data, null, 4)}</span>
+        sectionData = getReportSectionsData(
+            data,
+            reportParameters.mappedConfiguration,
+            reportParameters.currentPeriod
+        )
     }
 
     return (
-        <Button
-            primary
-            onClick={() => {
-                generateReport()
-            }}
-        >
-            Generate
-        </Button>
+        <>
+            {/* section one content. this needs to be improved */}
+
+            <div
+                className="report-preview report-preview-container"
+                style={{ margin: '26px' }}
+            >
+                <div
+                    style={{
+                        backgroundColor: '#5b92e5',
+                        padding: '5px',
+                        color: '#fff', // Assuming 'fer' is a valid color value
+                        fontSize: '20px',
+                        // margin: '20px'
+                    }}
+                >
+                    <p> DOMAIN 1 - COMPLETENESS OF REPORTING </p>
+                </div>
+                <p>1a: Completeness of facility reporting</p>
+                <p>
+                    The percentage of expected reports that have been entered
+                    and completed.
+                </p>
+                <Table>
+                    <TableHead>
+                        <TableRowHead>
+                            <TableCellHead>Data set</TableCellHead>
+                            <TableCellHead>Quality threshold</TableCellHead>
+                            <TableCellHead>Overall score</TableCellHead>
+                            <TableCellHead colSpan="3">
+                                Regions with divergent score
+                                <TableCellHead>Number</TableCellHead>
+                                <TableCellHead>Percentage</TableCellHead>
+                                <TableCellHead>Name</TableCellHead>
+                            </TableCellHead>
+                        </TableRowHead>
+                    </TableHead>
+                    <TableBody>
+                        {sectionData
+                            ? Object.values(
```

Comment:
since this is an array, taking `Object.values(sectionData.section1.section1A).map` doesn't do anything different than `sectionData.section1.section1A.map` (except to make it a bit harder to understand)